cumulus/zombienet/zombienet-sdk/tests/bootnodes/mod.rs

```diff
@@ -0,0 +1,113 @@
+	let images = zombienet_sdk::environment::get_images_from_env();
+	log::info!("Using images: {images:?}");
+	NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			let r = r
+				.with_chain("rococo-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				// Not strictly necessary for the test, but to keep it consistent
+				// with the parachain part we also pass `--no-mdns` to the relaychain.
+				.with_default_args(vec!["-lparachain=debug".into(), "--no-mdns".into()])
+				// Have to set a `with_node` outside of the loop below, so that `r` has the right
+				// type.
+				.with_node(|node| node.with_name("validator-0"));
+			(1..3).fold(r, |acc, i| acc.with_node(|node| node.with_name(&format!("validator-{i}"))))
+		})
+		.with_parachain(|p| {
+			p.with_id(1000)
+				.with_default_command("polkadot-parachain")
+				.with_default_image(images.cumulus.as_str())
+				.with_chain("asset-hub-rococo-local")
+				// Do not put bootnodes into the chain-spec nor command line arguments.
+				.without_default_bootnodes()
+				// Disable mdns to rely only on DHT bootnode discovery mechanism.
+				.with_default_args(vec![
+					"-lbootnodes=trace".into(),
+					"--no-mdns".into(),
+					"--discover-local".into(),
+					"--".into(),
+					"--no-mdns".into(),
+				])
+				.with_collator(|n| n.with_name("alpha"))
+				.with_collator(|n| n.with_name("beta"))
+		})
+		.with_global_settings(|global_settings| match std::env::var("ZOMBIENET_SDK_BASE_DIR") {
+			Ok(val) => global_settings.with_base_dir(val),
+			_ => global_settings,
+		})
+		.build()
+		.map_err(|e| {
+			let errs = e.into_iter().map(|e| e.to_string()).collect::<Vec<_>>().join(" ");
+			anyhow!("config errs: {errs}")
+		})
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn dht_bootnodes_test() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	let config = build_network_config().await?;
+	let spawn_fn = zombienet_sdk::environment::get_spawn_fn();
+	let mut network = spawn_fn(config).await?;
+
+	let alpha = network.get_node("alpha")?;
+
+	// Make sure the collators connect to each other.
+	alpha
+		.wait_metric_with_timeout("substrate_sync_peers", |count| count == 1.0, 300u64)
+		.await?;
+
+	// Make sure the DHT bootnode discovery was successful.
+	alpha
+		.wait_log_line_count_with_timeout(
+			".* Parachain bootnode discovery on the relay chain DHT succeeded",
+			false,
+			1,
+			30u64,
+		)
+		.await?;
+
+	log::info!(
+		"First two collators successfully connected via DHT bootnodes. \
+		 Waiting 150 secs for two sessions to pass."
+	);
+
+	// Wait for two sessions (2 min + 30 sec extra) and spawn a new collator to check the bootnode
+	// is also advertised with the new epoch key (republishing works).
+	tokio::time::sleep(std::time::Duration::from_secs(150)).await;
```

Comment:
We already have logic to wait for the next session here: [lib.rs](https://github.com/paritytech/polkadot-sdk/blob/524f9f40f450c27edbe61a838a5d6c6cf64a6cf9/cumulus/zombienet/zombienet-sdk-helpers/src/lib.rs#L240-L240), please use that. You can modify it to wait for N sessions. The method that exists there now can internally just call this with `N=1`.